### ISSUE TITLE
Support trackSiteSearch with mtm.push

### ISSUE THREAD

**Issue by Chardonneaur** [matomo-org/tag-manager#569]
Hello,
it is currently not possible to support the trackSiteSearch with mtm.push.
Is there any development planned about it?

Best regards,

**Comment by sgiehl** [comment]
@Chardonneaur Is this a tagmanager related request?

**Comment by Chardonneaur** [comment]
@sgiehl maybe it is more a documentation question. In fact, the differences between paq.push and mtm.push is not clear. Is mtm.push only tag manager related or can it be used with any other methods and will substitute all paq.push methods? My clients are saying that mtm.push work for most cases but not for the track site search.

**Comment by sgiehl** [comment]
I'd say mtm is only tag manager. I'll move the issue there, maybe someone with more tagmanager know how can answer that.

**Comment by snake14** [comment]
I wonder if this could potentially be related to an issue where [_paq.push doesn't always work right with MTM](https://github.com/matomo-org/tag-manager/issues/300). I don't think that `trackSiteSearch` will work with `_mtm.push`, but it should work with `_paq.push` while using our tag manager. If it doesn't, it would be related to that existing issue. Anything to add @AltamashShaikh ?

**Comment by AltamashShaikh** [comment]
@snake14 You are correct, we might need to create an additional tag/option in the existing code to support it or `_paq.push(['trackSiteSearch'keyword, category, searchCount])` can be executed as a custom HTML tag with a combination of Custom Variable having the other required params.

**Comment by tomper00** [comment]
I might have time look into a pull request for this next week. 
My suggestion is to add trackSiteSearch as an option in the Matomo Tag as mentioned here:
https://github.com/matomo-org/tag-manager/issues/570

Regarding the implementation the documentation for search suggests you do not track page views when you track search.
I can see the point from an analytics perspective, but are there any other things to think about there from a more technical perspective?

**Comment by AltamashShaikh** [comment]
cc @mattab for your view on this

**Comment by tomper00** [comment]
I just added some code that adds search tracking to the Matomo Tag
https://github.com/matomo-org/tag-manager/pull/575
@mattab

**Comment by Chardonneaur** [comment]
@tomper00 honestly, if you make it pass, for sure, I would like to interview you for my book dealing with this part.

**Comment by tomper00** [comment]
@Chardonneaur it definitely needs some documentation, since everything is kind of connected :)

**Comment by Chardonneaur** [comment]
@tomper00 I would be very interested into. In fact, so far, contributing thanks to translation, I get it, contributing thanks to developing, I get it. But now, how to contribute by coding to an existing plugin... question mark for me.

**Comment by heurteph-ei** [comment]
Hi @Chardonneaur 
It seems the MR #575 from @tomper00 hasn't been accepted (and closed). Then this enhancement will never be developed... 😢
It's a shame, because I don't think much additional work is needed.

**Comment by AltamashShaikh** [comment]
@heurteph-ei FYI, we closed #575 due to inactivity, but we have added efforts to solve this issue and its under prioritisation by our product team now to add this is out todo list.
